castoredc_api/client/castoredc_api_client.py

```diff
@@ -905,6 +1094,7 @@ async def async_get(self, url: str, params: list) -> list:
                         asyncio.as_completed(tasks),
                         total=len(tasks),
                         desc=f"Async Downloading {idx + 1}/{len(chunks)}",
+                        file=sys.stdout,
```

Comment:
This should probably be `sys.stderr` instead. Consider writing a script that outputs some information, which a user wishes to save to a file (or pipe to some other utility, like sed or awk). They would not want this spinner interfering with their output.

Conventionally, diagnostic information of any kind is printed to stderr, not stdout.

(Also, is `tqdm` smart enough not to print anything if stdin is not a TTY? Think printing spinner when a script is run as a cronjob or systemd unit, where output goes to the system journal/syslog)

Reply:
Alright, thanks for the heads up.
This was done to sync up the print() functions that notify the user of progress with the tqdm error-bars so they print neatly (proper newlines etc.).
Would an appropriate solution be to let everything print to std.err to notify the user of the download/upload progress?
This would output everything neatly to the correct channel for diagnostic information?

Reply:
>  This was done to sync up the print() functions that notify the user of progress with the tqdm error-bars so they print neatly (proper newlines etc.).
>  Would an appropriate solution be to let everything print to std.err to notify the user of the download/upload progress?

I was not aware there were other places where information was printed, but yes, converting all of these to use `stderr` is a good approach to take :+1: